Assignment6.Rmd

```diff
@@ -67,16 +67,46 @@ M2$predict1 <- predict(c.tree1, M2, type = "class")
 M2$predict2 <- predict(c.tree2, M2, type = "class")
 
 table(M2$certified, M2$predict1)
+(7790+24)/(2056+130+7790+24)
 
 table(M2$certified, M2$predict2)
+(3453+1184)/(896+1184+3453+4467)
+
+#The second tree has a lower error rate.
+
 
 ```
 
 ##Part III
 
 Choose a data file from the (University of Michigan Open Data Set)[https://github.com/bkoester/PLA/tree/master/data]. Choose an outcome variable that you would like to predict. Build two models that predict that outcome from the other variables. The first model should use raw variables, the second should feature select or feature extract variables from the data. Which model is better according to the cross validation metrics?
 
+According to the error rate, we choose the model one,which has a lower error rate.
 ```{r}
+library(dplyr)
+library(tidyverse)
+library(rpart)
+student.record<-read.csv("https://github.com/bkoester/PLA/blob/master/data/student.record.csv?raw=true",header = TRUE)
+
+#Model 1
+
+student.record.new<-select(student.record,SEX,MAJOR1_DESCR,HSGPA,LAST_ACT_ENGL_SCORE,LAST_ACT_MATH_SCORE,LAST_ACT_READ_SCORE,LAST_ACT_SCIRE_SCORE,LAST_ACT_COMP_SCORE)
+student.record.new<-na.omit(student.record.new)
+c.tree3<-rpart(SEX~LAST_ACT_ENGL_SCORE+LAST_ACT_MATH_SCORE+LAST_ACT_READ_SCORE+LAST_ACT_SCIRE_SCORE,method="class",data=student.record.new)
+student.record.new$predict1 <- predict(c.tree3,student.record.new, type = "class")
+table(student.record.new$SEX, student.record.new$predict1)
+#error rate=36.14%
+(11354+4400)/(11354+4400+18256+8576)
+
+
+#Model 2 
+c.tree4<-rpart(SEX~LAST_ACT_READ_SCORE+LAST_ACT_SCIRE_SCORE+HSGPA,method="class",data=student.record.new)
+
+
+student.record.new$predict2 <- predict(c.tree4,student.record.new, type = "class")
+table(student.record.new$SEX, student.record.new$predict2)
+#error rate=39.14%
+(10502+6168)/(16488+6168+10502+9428)
```

Comment:
Nice work. So based on the error rates and other measures from your result, which model would you choose and why?